### ISSUE TITLE
no manifest found

### ISSUE THREAD

**Issue by Jefe2** [fmartinou/whats-up-docker#202]
Since early December, I have gotten an error for any Linuxserver container that I have. For example, when `apprise-api` is watched, the following error is logged:

`WARN whats-up-docker/watcher.docker.local: Error when processing (Unexpected error; no manifest found) (container=local_apprise-api)`

These containers were previously watched successfully.

Edit:
The same error is received whether using linuxserver (lscr.io) or Docker Hub registries.  My hardware is RPi 4 with 4GB RAM, using linux/arm64.

**Comment by abrbon** [comment]
I have exact same issue, using a RaspberryPi 4B 4GB RAM with latest Docker (20.10.23), Docker-compose (v2.15.1) 
The message appears on containers from pihole and homeassistant

```
20:33:00.163  WARN whats-up-docker/watcher.docker.local: Error when processing some containers (Cannot read properties of undefined (reading 'tag'))
20:33:00.163  INFO whats-up-docker/watcher.docker.local: Cron finished (0 containers watched, 0 errors, 0 available updates)
20:33:04.211  WARN whats-up-docker/watcher.docker.local: Error when processing (Unexpected error; no manifest found) (container=local_duplicati)
20:39:32.508  WARN whats-up-docker/watcher.docker.local: Error when processing (Unexpected error; no manifest found) (container=local_homeassistant)
20:39:32.518  WARN whats-up-docker/watcher.docker.local: Error when processing some containers (Cannot read properties of undefined (reading 'tag'))
20:39:36.499  WARN whats-up-docker/watcher.docker.local: Error when processing (Unexpected error; no manifest found) (container=local_duplicati)

```

But other containers do not have this warning. For example Nextcloud, Nginx ProxyManager, Netdata.

I have a Docker hub free account (so not using the anonymous option)

**Comment by abrbon** [comment]
Just tried the new 6.0.2 update but i still have the same issues with my 2 containers, Homeassistant and Duplicati.
It is still giving me the exact same warnings.

I have also tried to completely remove the 2 containers, including the images and recreated them from my compose file in Portainer with Stacks. But that doesn't make any difference.

Tried couple of different tags, like stable and latest and even specific arm64v8-latest for the duplicati container but it doesn't make any difference.

Any thoughts?

**Comment by jimmz86** [comment]
same here with images from linuxserver image: linuxserver/heimdall:latest

**Comment by fmartinou** [comment]
Are you all experiencing this error on `arm` platforms only (Raspberry pi...) or also on `amd64 / x86` platforms?

**Comment by jimmz86** [comment]
raspbian/arm64 RPI4 here

**Comment by Jefe2** [comment]
Appears to be `arm` only. I have a `arm64` system and a `amd64` system.  Only the `arm` system has the issue.

**Comment by abrbon** [comment]
I can also confirm it is only with ARM on my Raspi 4.
I also have Watchtower installed which i used before i discovered WUD and with Watchtower it works just fine on the same Raspi 4.

And on my other Docker server installed on Debian Linux with amd64 WUD works fine also.

**Comment by fmartinou** [comment]
Thank you all for your feedbacks. 👍 

I don't know yet what's going wrong but in the meantime, I prepared a version with additional debug logs to help me figure out what happens.

Can you run the version
```
fmartinou/whats-up-docker:fix__202_no_manifest_found
```
with debug logs enabled (`WUD_LOG_LEVEL=debug`) then post the logs here?

Thank you!

**Comment by Jefe2** [comment]
Thank you for looking into this issue.  I’m away for a couple days, but I can post logs early next week.

**Comment by arjandriel** [comment]
Hello, hereby some logs as you requested. I am also running on a RPI 4. I hope this can help you solve this. This is the log from one complete refresh.

[_whats-up-docker_logs.txt](https://github.com/fmartinou/whats-up-docker/files/11070266/_whats-up-docker_logs.txt)

**Comment by fmartinou** [comment]
Thank you very much for the logs 👍 .

I understand now what's going wrong.

For example, for `homeassistant`, according to your logs, the local image has the following properties:

```yaml
arch: arm64
os: linux
variant: undefined
```

The list of the manifests that are received from the registry for this image are
```yaml
- mediaType: application/vnd.docker.distribution.manifest.v2+json
  size: 6388
  digest: sha256:74cd287efbbcb3497da6a922cfe5159d2dfb2672ce3c20f268b760334e9bcab8
  platform:
    architecture: arm64
    os: linux
    variant: v8
- mediaType: application/vnd.docker.distribution.manifest.v2+json
  size: 6388
  digest: sha256:58dcc33ee408a92c41c57e4f853612f8f8001d98ed8d1d3af0532d25fbaa995a
  platform:
    architecture: amd64
    os: linux
- mediaType: application/vnd.docker.distribution.manifest.v2+json
  size: 6389
  digest: sha256:5d7cc8c050fe50716e1e704d04edc5f62aae91d720d44030d90dc46831509792
  platform:
    architecture: arm
    os: linux
    variant: v6
- mediaType: application/vnd.docker.distribution.manifest.v2+json
  size: 6388
  digest: sha256:4bb4a4f7205b5d5c3dc801fbbd90490033e36d8ee394df8511e6a19c1c6a1bdd
  platform:
    architecture: arm
    os: linux
    variant: v7
- mediaType: application/vnd.docker.distribution.manifest.v2+json
  size: 6388
  digest: sha256:0e53b6521e5355570af2846884a073de95117272f00ec14869cea988e6dda840
  platform:
    architecture: '386'
    os: linux
```

The manifest  that should be matching is
```
mediaType: application/vnd.docker.distribution.manifest.v2+json
size: 6388
digest: sha256:74cd287efbbcb3497da6a922cfe5159d2dfb2672ce3c20f268b760334e9bcab8
platform:
  architecture: arm64
  os: linux
  variant: v8
```

But it's not because the variant is not the same as the one on the local image (`undefined !== 'v8'`)

=> I'll change the algo to be more tolerant and filter on the variant only if it's defined AND that multiple manifests are matching.

**Comment by fmartinou** [comment]
I pushed a fix; can you test it?
It's still the tag `fmartinou/whats-up-docker:fix__202_no_manifest_found`
(don't forget to pull it first to be sure to run the latest built version)

**Comment by arjandriel** [comment]
Hi, i updated the container, all containers look to be recognized. See attachted log.

[_whats-up-docker_logs (1).txt](https://github.com/fmartinou/whats-up-docker/files/11071738/_whats-up-docker_logs.1.txt)

**Comment by fmartinou** [comment]
Great! 🎉

**Comment by fmartinou** [comment]
Released as `6.1.1`

**Comment by jimmz86** [comment]
thats it works! all errors are gone :) thanks @fmartinou
